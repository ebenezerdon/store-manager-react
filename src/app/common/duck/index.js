export {
  doFetchCurrentUser,
  doFetchAllUsers,
  doFetchSaleRecord,
  doFetchProducts
} from './operations';
export { default as fetchAllDataReducer } from './reducers';
export { default as constants } from './constants';
export { default as types } from './types';
export { default as actions } from './actions';
