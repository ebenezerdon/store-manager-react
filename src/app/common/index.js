export { fetchAllDataReducer } from './duck';
export { constants } from './duck';
export { EllipsisLoaderComponent } from './loaders';

export {
  doFetchCurrentUser,
  doFetchAllUsers,
  doFetchSaleRecord,
  doFetchProducts
} from './duck';
