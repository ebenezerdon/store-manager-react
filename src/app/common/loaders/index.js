export { default as EllipsisLoaderComponent } from './EllipsisLoaderComponent';
