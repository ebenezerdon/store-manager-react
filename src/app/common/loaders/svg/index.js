export { default as ellipsisLoader } from './EllipsisLoader.svg';