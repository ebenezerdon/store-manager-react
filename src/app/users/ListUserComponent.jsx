import React from 'react';
import lifecycle from 'react-pure-lifecycle';
import { Link } from 'react-router-dom';
import { constants } from '../common';
import './listUser.css';

const methods = {
  componentDidMount({
    fetchStoreData,
    userData
  }) {
    if (!userData.allUsers) {
      fetchStoreData()
    }
  }
};

const ListUserComponent = ({
  userData,
  fetchAllUserState,
}) => {
  const listUser = () => {
    const { allUsers } = userData;
    return (
      allUsers.map(user => (
        <>
          <tr>
            <td><Link to={`/user/${user.id}`}>{user.fullname}</Link></td>
            <td>{user.emailaddress}</td>
            <td>{user.phonenumber}</td>
            <td>{user.role}</td>
          </tr>
        </>
      ))
    )
  }

  return (
    <>
      <Link to="/new/user">
          <button id="add-product-btn">Add New User</button>
      </Link>
      <table id="listUserTable" className="table table-striped">
        <thead>
          <tr>
            <th>Name</th>
            <th>Email Address</th>
            <th>Phone Number</th>
            <th>Role</th>
          </tr>
        </thead>
        <tbody>{userData.allUsers && listUser()}</tbody>
      </table>
    </>
  )
}

export default lifecycle(methods)(ListUserComponent);
