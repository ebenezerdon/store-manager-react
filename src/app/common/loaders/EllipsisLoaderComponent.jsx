import React from 'react';
import { ellipsisLoader } from './svg';

const EllipsisLoaderComponent = () => {
  return <img className="ellipsis" src={ellipsisLoader} alt="loader" />;
};

export default EllipsisLoaderComponent;
