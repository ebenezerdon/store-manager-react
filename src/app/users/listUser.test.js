import React from 'react';
import { shallow, mount } from 'enzyme';
import { EllipsisLoaderComponent } from '../common/loaders'
import ListUserComponent from './ListUserComponent';
import {
  mapDispatchToProps,
  mapStateToProps,
} from './ListUserContainer';

describe('ListUser Container', () => {
  it('should show initial state', () => {
    const initialState = {
      fetchAllData: { cart: { Data: ['Data'] } },
      errorMessage: '',
      checkout: {
        checkoutState: '',
      },
    };
    expect(mapStateToProps(initialState).fetchAllUserState).not.toBe(false);
  });

  it('should dispatch action', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).fetchStoreData();
    expect(typeof dispatch.mock.calls[0][0]).toEqual('function');
  });
});

describe('List user component', () => {
  it('should render the component', () => {
    const fetchStoreData = jest.fn();
    const userData = {allUsers: []};
    const allUsers = {};
    const component = shallow(
      <ListUserComponent
        {...{ fetchStoreData, userData, allUsers }}
      />,
    );
    expect(component.exists()).toBe(true);
    expect(component).toMatchSnapshot();
  });
});
