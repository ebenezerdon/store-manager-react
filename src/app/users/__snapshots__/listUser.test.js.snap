// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List user component should render the component 1`] = `
<ListUserComponent
  allUsers={Object {}}
  fetchStoreData={[MockFunction]}
  userData={
    Object {
      "allUsers": Array [],
    }
  }
/>
`;
